### PR TITLE
[PLATFORM-509]: [Auth0Ex] More generic Absinthe plug

### PR DESCRIPTION
**Pull request by cpiemontese**
https://prima-assicurazioni-spa.myjetbrains.com/youtrack/issue/PLATFORM-509

This PR tries to make the Absinthe plug more generic by using a map instead of a structure for the context

### CODE REVIEW

**Comment by cpiemontese** [comment]
@enerdgumen one thing that is unclear to me is whether we want everything under `:context` or I can just put `:auth0` at the top level and call it a day

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,6 +7,10 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Next]
 
+### Changed
+
+- Absinthe plugs do not use a custom context structor but instead use a map
```
Comment (mbusi) [inline CHANGELOG.md:12]:
```suggestion
- Absinthe plugs use a map instead of custom struct as context
```

**Review by mbusi — COMMENTED** [review]
Why is this change needed? :o

**Comment by cpiemontese** [comment]
> Why is this change needed? :o

You can find a reasoning https://github.com/primait/urania/pull/481#discussion_r898776553, but TL;DR using a custom Context prevents the plug from being used with other plugs which add stuff to the context

**Comment by mbusi** [comment]
> > Why is this change needed? :o
> 
> You can find a reasoning [primait/urania#481 (comment)](https://github.com/primait/urania/pull/481#discussion_r898776553), but TL;DR using a custom Context prevents the plug from being used with other plugs which add stuff to the context

Yeah sorry I am dumb and didn't check the comment that was linked in the youtrack issue -.-"
BTW, even though I get the point of the requirement I don't think you need to stop using a struct, so I guess using `assign_context` is fine (and the right way to do it), but your `auth0` value inside the map absinthe context could still be a struct

**Comment by cpiemontese** [comment]
You are right, the top-level `context` needs to be "structless" but not the internal `auth0`. I guess I went overboard 😅

**Review by enerdgumen — APPROVED** [review]
Perfect!

**Pull request merged**